tests/data_snack/connections/test_memcached.py

```diff
@@ -0,0 +1,54 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from data_snack.connections.memcached import MemcachedConnection
+
+
+@pytest.fixture
+def connection() -> MemcachedConnection:
+    """A `MemcachedConnection` with mocked connection to the actual database."""
+    return MemcachedConnection(connection=MagicMock())
+
+
+def test_get(connection: MemcachedConnection) -> None:
+    """Testing getting a single value based of the provided key."""
+    connection.connection.get.return_value = "value"
+
+    result = connection.get("key")
+    assert result == "value"
+    connection.connection.get.assert_called_with("key")
+
+
+def test_set(connection: MemcachedConnection) -> None:
+    """Testing setting a single value based for the provided key."""
+    connection.connection.set.return_value = "key"
+
+    result = connection.set("key", "value")
+    assert result == "key"
+    connection.connection.set.assert_called_with("key", "value")
+
+
+def test_get_many(connection: MemcachedConnection) -> None:
+    """Testing getting multiple values based on a provided list of keys."""
+    connection.connection.get_many.return_value = {"key1": "value1", "key2": "value2"}
+
+    result = connection.get_many(["key1", "key2"])
+    assert result == {"key1": "value1", "key2": "value2"}
+    connection.connection.get_many.assert_called_with(["key1", "key2"])
+
+
+def test_set_many(connection: MemcachedConnection) -> None:
+    """Testing setting multiple values. Values are provided in a form of a dictionary."""
+    connection.connection.set_many.return_value = []
+
+    result = connection.set_many({"key1": "value1", "key2": "value2"})
+    assert set(result) == {"key1", "key2"}
+    connection.connection.set_many.assert_called_with({"key1": "value1", "key2": "value2"})
+
+
+@pytest.mark.xfail(raises=NotImplemented)
```

Comment:
Why skip this test since we could assert  
`with pytest.raises(NotImplementedError):`
and change implementation like this  
```
    def keys(self, pattern: Text) -> List[Text]:
        raise NotImplementedError
```

Reply:
I tried that, but for some reason it causes pytest error

```
INTERNALERROR>   File "../.pyenv/versions/3.9.14/envs/data-snack/lib/python3.9/site-packages/_pytest/skipping.py", line 276, in pytest_runtest_makereport
INTERNALERROR>     if raises is not None and not isinstance(call.excinfo.value, raises):
INTERNALERROR> TypeError: isinstance() arg 2 must be a type or tuple of types
````

Reply:
Changed to `NotImplementedError` and it works correctly